probe/endpoint/dns_snooper_linux_amd64.go

```diff
@@ -203,6 +207,20 @@ func (s *DNSSnooper) run() {
 	}
 }
 
+// handleDecodeError logs errors up to the maximum allowed count
+func (s *DNSSnooper) handleDecodingError(err error) {
+	str := err.Error()
+	count := s.decodingErrorCounts[str]
+	count++
+	s.decodingErrorCounts[str] = count
```

Comment:
Ideally we should use a size-limited cache with time-based expiration on keys. That's overkill though. We are running into this problem quite often, so I may consider creating a library for this.